components/ws-manager/pkg/manager/manager.go

```diff
@@ -671,6 +671,12 @@ func podRunning(clientset client.Client, podName, namespace string) wait.Conditi
 					return true, nil
 				}
 			}
+			// if we failed to mount volume, we need to re-create the pod
+			// ref: https://github.com/gitpod-io/gitpod/issues/13353
+			// ref: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/608
+			if strings.Contains(pod.Status.Reason, "MountVolume.MountDevice failed for volume") {
```

Comment:
Could we have a log when we-manager detects the issue happens? So we could know the number of times we hit this issue.

Reply:
We already have it. On line 361.

Reply:
Since we haven't reproduced it yet, and also we did not dump the pod YAML manifest when the last time Toru was able to reproduce. I worried that the ws-manager did not recreate the pod because one of the conditions mismatch
https://github.com/gitpod-io/gitpod/blob/e3d962295e9529d3c40bce0008e1e08295f0dc05/components/ws-manager/pkg/manager/manager.go#L703-L707